internal/pkg/kube/destination_rule_reconciler.go

```diff
@@ -46,36 +50,73 @@ func (r *DestinationRuleReconciler) GetTypeUrl() string {
 }
 
 func (r *DestinationRuleReconciler) Reconcile(ctx context.Context) error {
-	dr := r.cf.GetDestinationRules()
-	if dr == nil {
+	destinationRules := r.cf.DestinationRules()
+	if len(destinationRules) == 0 {
 		return nil
 	}
 
-	kind := "DestinationRule"
-	apiVersion := "networking.istio.io/v1alpha3"
-	newDR, err := r.client.Istio().NetworkingV1alpha3().DestinationRules(dr.GetNamespace()).Apply(ctx, &networkingv1alpha3.DestinationRuleApplyConfiguration{
-		TypeMetaApplyConfiguration: applyconfigurationv1.TypeMetaApplyConfiguration{
-			Kind:       &kind,
-			APIVersion: &apiVersion,
-		},
-		ObjectMetaApplyConfiguration: &applyconfigurationv1.ObjectMetaApplyConfiguration{
-			Name:      &dr.Name,
-			Namespace: &dr.Namespace,
-		},
-		Spec:   &dr.Spec,
-		Status: nil,
-	}, metav1.ApplyOptions{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       kind,
-			APIVersion: apiVersion,
-		},
-		Force:        true,
-		FieldManager: "federation-controller",
+	destinationRulesMap := make(map[types.NamespacedName]*v1alpha3.DestinationRule, len(destinationRules))
+	for _, dr := range destinationRules {
+		destinationRulesMap[types.NamespacedName{Namespace: dr.Namespace, Name: dr.Name}] = dr
+	}
+
+	oldDestinationRules, err := r.client.Istio().NetworkingV1alpha3().DestinationRules(metav1.NamespaceAll).List(ctx, metav1.ListOptions{
+		LabelSelector: metav1.FormatLabelSelector(&metav1.LabelSelector{
+			MatchLabels: map[string]string{"federation.istio-ecosystem.io/peer": "todo"},
+		}),
 	})
 	if err != nil {
-		return fmt.Errorf("failed to apply destination rule: %w", err)
+		return fmt.Errorf("failed to list destination rules: %w", err)
+	}
+	oldDestinationRulesMap := make(map[types.NamespacedName]*v1alpha3.DestinationRule, len(oldDestinationRules.Items))
+	for _, se := range oldDestinationRules.Items {
+		oldDestinationRulesMap[types.NamespacedName{Namespace: se.Namespace, Name: se.Name}] = se
+	}
```

Comment:
```suggestion
	for _, dr := range oldDestinationRules.Items {
		oldDestinationRulesMap[types.NamespacedName{Namespace: dr.Namespace, Name: dr.Name}] = dr
	}
```

Reply:
Copy-paste... Thanks :)

Reply:
Oh yeah, I was looking for this one as I noticed it after first read a few days ago :D But I failed after a quick scan so I assumed it was fixed in the meantime :) Thanks @eoinfennessy